client/db/src/lib.rs

```diff
@@ -1347,11 +1346,6 @@ impl<Block: BlockT> Backend<Block> {
 			}
 
 			if number.is_zero() {
-				transaction.set_from_vec(
```

Comment:
And this?

Reply:
This is wrong because we don't want to re-finalize genesis when restarting interrupted fast sync. When doing fast/warp  sync we don't really commit genesis state, because it won't be properly pruned when a recent state is imported. So genesis is not committed so that it does not leave any junk in the database. However, when the fast sync is restarted, we need to populate genesis again, because babe needs to read configuration from it. When this happens we also have some of the headers synced and finalized so we don't want to re-finalize genesis again. Basically the changes here remove unconditional genesis finalization and make sure it is finalized only when being actually committed for the first time.

Reply:
Ty for the explanation.